contracts/examples/PairFlash.sol

```diff
@@ -0,0 +1,158 @@
+        uint256 amount0Min = LowGasSafeMath.add(decoded.amount0, fee0);
+
+        // call exactInputSingle for swapping token1 for token0 in pool w/fee2
+        uint256 amountOut0 =
+            swapRouter.exactInputSingle(
+                ISwapRouter.ExactInputSingleParams({
+                    tokenIn: token1,
+                    tokenOut: token0,
+                    fee: decoded.poolFee2,
+                    recipient: address(this),
+                    deadline: block.timestamp + 200,
+                    amountIn: decoded.amount1,
+                    amountOutMinimum: amount0Min,
+                    sqrtPriceLimitX96: 0
+                })
+            );
+
+        // call exactInputSingle for swapping token0 for token 1 in pool w/fee3
+        uint256 amountOut1 =
+            swapRouter.exactInputSingle(
+                ISwapRouter.ExactInputSingleParams({
+                    tokenIn: token0,
+                    tokenOut: token1,
+                    fee: decoded.poolFee3,
+                    recipient: address(this),
+                    deadline: block.timestamp + 200,
+                    amountIn: decoded.amount0,
+                    amountOutMinimum: amount1Min,
+                    sqrtPriceLimitX96: 0
+                })
+            );
+
+        // end up with amountOut0 of token0 from first swap and amountOut1 of token1 from second swap
+        uint256 amount0Owed = LowGasSafeMath.add(decoded.amount0, fee0);
+        uint256 amount1Owed = LowGasSafeMath.add(decoded.amount1, fee1);
+
+        TransferHelper.safeApprove(token0, address(this), amount0Owed);
+        TransferHelper.safeApprove(token1, address(this), amount1Owed);
+
+        if (amount0Owed > 0) pay(token0, address(this), msg.sender, amount0Owed);
+        if (amount1Owed > 0) pay(token1, address(this), msg.sender, amount1Owed);
+
+        // if profitable pay profits to payer
+        if (amountOut0 > amount0Owed) {
+            uint256 profit0 = LowGasSafeMath.sub(amountOut0, amount0Owed);
+
+            TransferHelper.safeApprove(token0, address(this), profit0);
+            pay(token0, address(this), decoded.payer, profit0);
+        }
+        if (amountOut1 > amount1Owed) {
+            uint256 profit1 = LowGasSafeMath.sub(amountOut1, amount1Owed);
+            TransferHelper.safeApprove(token0, address(this), profit1);
+            pay(token1, address(this), decoded.payer, profit1);
+        }
+    }
+
+    //fee1 is the fee of the pool from the initial borrow
+    //fee2 is the fee of the first pool to arb from
+    //fee3 is the fee of the second pool to arb from
+    struct FlashParams {
+        address token0;
+        address token1;
+        uint24 fee1;
+        uint256 amount0;
+        uint256 amount1;
+        uint24 fee2;
+        uint24 fee3;
+    }
+    // fee2 and fee3 are the two other fees associated with the two other pools of token0 and token1
+    struct FlashCallbackData {
```

Comment:
Another nit! I'd stick this struct right above the `uniswapV3FlashCallback` function since that's where it's most relevant. Confusing to need to dig for it.